pkg/standalone-utils/contracts/LPOracleBase.sol

```diff
@@ -276,6 +307,30 @@ abstract contract LPOracleBase is ILPOracleBase, AggregatorV3Interface {
         return feedTokenDecimalScalingFactors;
     }
 
+    /**
+     * @notice Ensure that the sequencer is up and, if there was a recent outage, that the grace period has passed.
+     * @dev Reverts if the sequencer is down or the grace period has not expired.
+     */
+    function _ensureSequencerUptime() internal view {
+        if (address(_sequencerUptimeFeed) == address(0)) {
+            return; // No sequencer check needed (L1 or other networks)
+        }
+
+        // Check the status of the uptime feed.
+        try _sequencerUptimeFeed.latestRoundData() returns (uint80, int256 answer, uint256 startedAt, uint256, uint80) {
+            if (answer == _SEQUENCER_STATUS_DOWN) {
+                revert SequencerDown();
+            }
+
+            // solhint-disable-next-line not-rely-on-time
+            if (block.timestamp - startedAt < _uptimeGracePeriod) {
+                revert GracePeriodNotOver();
+            }
+        } catch {
+            revert SequencerFeedUnavailable();
```

Comment:
Is there any known reason why these contracts would revert, or is this just defensive programming?